_optimisations/python/numpy-array-resize.md

```diff
@@ -38,24 +41,27 @@ def array_resize():
     for i in range(1, N):
         ar.resize(i+1)
         ar[i] = i
-        
+
 repeats = 1000
-print(f"array_resize: {timeit(array_resize, number=repeats):.2f}ms")
-print(f"list_append: {timeit(list_append, number=repeats):.2f}ms")
 print(f"list_comprehension: {timeit(list_comprehension, number=repeats):.2f}ms")
+print(f"list_append: {timeit(list_append, number=repeats):.2f}ms")
+print(f"array_resize: {timeit(array_resize, number=repeats):.2f}ms")
 ```
 
 Resizing a NumPy array was 8x slower than a list, and 13x slower than list comprehension.
 
 ```output
-array_resize: 82.66ms
-list_append: 9.68ms
 list_comprehension: 6.29ms
+list_append: 9.68ms
+array_resize: 82.66ms
 ```
 
 ## The Technical Detail
 
 Resizing an array, allocates a new buffer in memory of the required size, copies the data across, and de-allocates the old storage.
 
-In comparison a list, whilst backed by an array, performs greedy resizes. The length of the list visible to the user, does not reflect the length of the internal array. This allows it to simply store new items and increase it's length counter when appending. Only occasionally, does it need to perform an additional greedy resize. This greatly improves the append performance, at the cost of a slight memory overhead.
-
+In comparison a list, whilst backed by an array, performs greedy resizes.
+The length of the list visible to the user does not reflect the length of the internal array.
+This allows it to simply store new items and increase its length-counter when appending.
+Only occasionally, does it need to perform an additional greedy resize.
+This greatly improves the append performance, at the cost of a slight memory overhead.

```

Comment:
For these reasons, it is more computationally efficient to use lists than numpy arrays [if/unless reasons...]
```suggestion
This greatly improves the append performance, at the cost of a slight memory overhead.

For these reasons, it is more computationally efficient to use a list rather than a NumPy array when the required length is unknown or changes frequently, since the latter would necessitate frequent and costly calls to `resize()`.
```

Based on your suggestion, I see no harm in adding a concluding sentence.